src/dvc_data/fs.py

```diff
@@ -71,23 +71,38 @@ def open(  # type: ignore
         return fs.open(fspath, mode=mode, encoding=encoding)
 
     def ls(self, path, detail=True, **kwargs):
-        info = self.info(path)
-        if info["type"] != "directory":
-            return [info] if detail else [path]
-
         root_key = self._get_key(path)
-        try:
-            entries = [
-                self.path.join(path, name) if path else name
-                for name in self.index.ls(prefix=root_key)
-            ]
-        except KeyError as exc:
-            raise FileNotFoundError from exc
+        info = self.index.info(root_key)
+        if info["type"] != "directory":
+            if detail:
+                info["name"] = path
+                return [info]
+            else:
+                return [path]
 
+        self.index._ensure_loaded(root_key)  # pylint: disable=protected-access
```

Comment:
Oh, so it is used outside of index. Sorry, I didn't notice it before. Then should be public, obviously.